docs/monitoring.md

```diff
@@ -0,0 +1,35 @@
+# Monitoring
+
+This document describes how to get [Sentry](https://sentry.io/) (the Epic
+application monitoring provider) set up for error, performance, and replay
+monitoring.
+
+> **NOTE**: this is an optional step and only needed if you want monitoring in
+> production.
+
+## SaaS vs Self-Hosted
+
+Sentry offers both a [SaaS solution](https://sentry.io/) and
+[self-hosted solution](https://develop.sentry.dev/self-hosted/). This guide
+assumes you are using SaaS but the guide still works with self-hosted with a few
+modifications.
+
+## Signup
+
+You can sign up for Sentry and create a Remix project from visting
+[this url](https://sentry.io/signup/?project_platform=javascript-remix) and
+filling out the signup form.
+
+## Onboarding
+
+Once you see the onboarding page which has the DSN, copy that somewhere (this
+becomes `SENTRY_DSN`). Then click
+[this](https://sentry.io/orgredirect/settings/:orgslug/developer-settings/new-internal/)
+to create an internal integration. Give it a name and add the scope for
+`Releases:Admin`. Press Save, find the auth token at the bottom of the page, and
+copy that to secure location (this becomes `SENTRY_AUTH_TOKEN`). Then vist the
+organization settings page and copy that organization slug (`SENTRY_ORG_SLUG`).
+
+Now, set the secrets in Fly.io:
+
+fly secrets set SENTRY_DSN=<your_dsn> SENTRY_AUTH_TOKEN=<your_auth_token> SENTRY_ORG=<your_org_slug> SENTRY_PROJECT=javascript-remix
```

Comment:
Is "javascript-remix" really supposed to hard coded? If so, maybe we should just put that in the Dockerfile?

````suggestion
```sh
fly secrets set SENTRY_DSN=<your_dsn> SENTRY_AUTH_TOKEN=<your_auth_token> SENTRY_ORG=<your_org_slug> SENTRY_PROJECT=javascript-remix
```
````

Reply:
@kentcdodds Yea it is supposed to be hardcoded because that's the name of the project that will get created, at least if you do it through new user onboarding (we don't ask for the project name there). Though if you change the project name then you'll have to make a change here.

Reply:
Oh I see, so users can put whatever value they want here and then reports will be associated to this project? Should this be set differently based on environment? We could do something for this in the setup script where we determine the app name.

Reply:
@kentcdodds  The name has to match what you have in Sentry, but the default project name for a Remix project is `javascript-remix`. You don't need a different project per env as Sentry will automatically pick up the `NODE_ENV` environment variable and tag that for each event. Though it's perfectly valid to make a separate project if you want as well, people do it both ways.